CRM/Core/BAO/OptionValue.php

```diff
@@ -234,7 +234,7 @@ public static function add(&$params, $ids = array()) {
 
     $optionValue->id = CRM_Utils_Array::value('optionValue', $ids);
     $optionValue->save();
-    CRM_Core_PseudoConstant::flush();
+    CRM_Core_OptionGroup::flushAll();
```

Comment:
Calling CRM_Core_PseudoConstant::flush() with no params simply results in CRM_Core_OptionGroup::flushAll() being called -so this is for clarity

Reply:
No, that's not all it does. It also clears the CRM_Core_PseudoConstant::$cache variable.

Reply:
It looks like that is not a real property though. I can't see anywhere that self::cache is set - it seems to me that either you call the function with the implicit param 'cache' and it does nothing in the first bit because CRM_Core_PseudoConstant::cache does not exist AND it does an OptionGroup flush in the second part. Or you pass in a parameter to flush & it clears that and not the optionGroup

```
  /**
   * Flush given pseudoconstant so it can be reread from db.
   * nex time it's requested.
   *
   *
   * @param bool|string $name pseudoconstant to be flushed
   */
  public static function flush($name = 'cache') {
    if (isset(self::$$name)) {
      self::$$name = NULL;
    }
    if ($name == 'cache') {
      CRM_Core_OptionGroup::flushAll();
    }
  }
```

(It's called from 3 places outside unit tests - including this one)

Reply:
One of those 3 is CRM_Utils_PseudoConstant::flushAll(); - which DOES flush all properties that are set. The last is OptionValue::delete

Reply:
The variable is used by `CRM_Core_PseudoConstant::get()` so almost all pseudoconstant lookups rely on it.

Reply:
OK - I can see it now - I'll flip that back